### PR TITLE
added support for gif images

### LINKED ISSUE #48: Add support for GIF
**Describe the bug**

I install the package with Conda, Python 3.8 and PIP. I run it on a GIF image and get this error:

```
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/oemer/bin/oemer", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 280, in main
    mxl_path = extract(args)
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 131, in extract
    staff, symbols, stems_rests, notehead, clefs_keys = generate_pred(str(img_path), use_tf=args.use_tf)
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 51, in generate_pred
    staff_symbols_map, _ = inference(
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/inference.py", line 60, in inference
    image = Image.fromarray(image).convert("RGB")
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/PIL/Image.py", line 3084, in fromarray
    arr = obj.__array_interface__
AttributeError: 'NoneType' object has no attribute '__array_interface__'
```

I get the same error using `oemer[tf]`.

**Input Image**

I attach the PNG and the GIF version. Only the GIF version causes the error.
ss015
ss015

**Full Traceback**

```
2023-11-17 10:45:48 Extracting staffline and symbols
/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py:69: UserWarning: Specified provider 'CoreMLExecutionProvider' is not in available provider names.Available providers: 'TensorrtExecutionProvider, CUDAExecutionProvider, AzureExecutionProvider, CPUExecutionProvider'
  warnings.warn(
2023-11-17 10:45:50.437510351 [E:onnxruntime:Default, provider_bridge_ort.cc:1480 TryGetProviderInfo_CUDA] /onnxruntime_src/onnxruntime/core/session/provider_bridge_ort.cc:1193 onnxruntime::Provider& onnxruntime::ProviderLibrary::Get() [ONNXRuntimeError] : 1 : FAIL : Failed to load library libonnxruntime_providers_cuda.so with error: libcublasLt.so.11: cannot open shared object file: No such file or directory

2023-11-17 10:45:50.437708300 [W:onnxruntime:Default, onnxruntime_pybind_state.cc:747 CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Please reference https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements to ensure all dependencies are met.
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/oemer/bin/oemer", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 280, in main
    mxl_path = extract(args)
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 131, in extract
    staff, symbols, stems_rests, notehead, clefs_keys = generate_pred(str(img_path), use_tf=args.use_tf)
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/ete.py", line 51, in generate_pred
    staff_symbols_map, _ = inference(
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/oemer/inference.py", line 60, in inference
    image = Image.fromarray(image).convert("RGB")
  File "/home/ubuntu/miniconda3/envs/oemer/lib/python3.8/site-packages/PIL/Image.py", line 3084, in fromarray
    arr = obj.__array_interface__
AttributeError: 'NoneType' object has no attribute '__array_interface__'
```

**Command You Execute**

```
$ oemer ss015.gif 
```

### PR DESCRIPTION
**Pull request by miguelmorin**
I made a minimal change to support GIF images in two files (input and output). The workflow runs, although I haven't inspected the resulting XML file. Closes #48.

### CODE REVIEW

File: oemer/ete.py
Comment (BreezeWhite) [inline oemer/ete.py:1]: I would not recommend distinguishing the file format through the image path. Using the following approach would be better

```python
image = Image.open(img_path)
if image.format == "GIF":
   ...
```

File: oemer/ete.py [hunk 1]
```diff
@@ -142,7 +142,13 @@ def extract(args: Namespace) -> str:
             pickle.dump(data, open(pkl_path, "wb"))
 
     # Load the original image, resize to the same size as prediction.
-    image = cv2.imread(str(img_path))
+    if not str(img_path).endswith(".gif"):
+        image = cv2.imread(str(img_path))
+    else:
+        gif_image = Image.open(str(img_path)).convert('RGB')
+        gif_image = np.array(gif_image)
+        image = gif_image[:, :, ::-1].copy()
```
Comment (BreezeWhite) [inline oemer/ete.py:150]: Is there any specific reason to shuffle the RGB channel?

File: oemer/inference.py
Comment (BreezeWhite) [inline oemer/inference.py:1]: Same as above. Using `PIL.open(img_path).format` would be better.

**Review by BreezeWhite — CHANGES_REQUESTED** [review]
Hi, thanks for the PR. I have left some comments and please kindly review them.

File: oemer/ete.py
Comment (miguelmorin) [reply oemer/ete.py:1]: Thanks, I amended that.

File: oemer/ete.py (hunk 1 above)
Comment (miguelmorin) [reply oemer/ete.py:151]: PIL.Image() uses RGB, OpenCV uses BGR. I shuffle the channels to ensure the GIF image has the same order of channels as the others, which may matter in the conversion to grayscale where each channel has a different weight (but I haven't checked in the code whether and where this happens).

File: oemer/inference.py
Comment (miguelmorin) [reply oemer/inference.py:1]: Done, thanks.

File: oemer/ete.py (hunk 1 above)
Comment (BreezeWhite) [reply oemer/ete.py:151]: I see. Thanks for the explanation.

**Comment by BreezeWhite** [comment]
The modification looks good to me. Merging this into main branch. Thank you for the contribution again @miguelmorin .

**Pull request merged**